src/languageSupport/languages/sql/connection.ts

```diff
@@ -0,0 +1,240 @@
+import {ConnectionManager as AgnosticConnectionManager} from 'src/languageSupport/languages/agnostic/connection'
+
+// Types
+import {
+  DEFAULT_SQL_EDITOR_TEXT,
+  CompositionSelection,
+} from 'src/dataExplorer/context/persistance'
+import {
+  SelectableDurationTimeRange,
+  TagKeyValuePair,
+  TimeRange,
+} from 'src/types'
+
+// Utils
+import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
+import {rangeToInterval as buildTimeRange} from 'src/shared/utils/sqlInterval'
+import {notify} from 'src/shared/actions/notifications'
+import {compositionEnded} from 'src/shared/copy/notifications'
+
+const castType = (value, _typ = null) => {
+  // Outstanding tech debt.
+  // For the TagValue selector (Qx builder), flux composition, and rest of codebase.
+  // It only supports tagValues of type string.
+  return `'${value}'`
+}
+
+const groupedTagValues = (
+  tagValues: TagKeyValuePair[]
+): {[key: string]: string[]} =>
+  tagValues.reduce(
+    (acc, {key, value}) => ({
+      ...acc,
+      [key]: (acc[key] || []).concat([value]),
+    }),
+    {}
+  )
+
+export class ConnectionManager extends AgnosticConnectionManager {
+  private _timeRange: TimeRange = DEFAULT_TIME_RANGE
+
+  _couldBeFromComposition(change) {
+    return change.forceMoveMarkers == true
+  }
+
+  _editorChangeIsWithinComposition(change) {
+    if (!this._compositionRange) {
+      return
+    }
+    const {
+      start: {line: startLine},
+      end: {line: endLine},
+    } = this._compositionRange
+
+    const changeInBlock =
+      change.range.startLineNumber >= startLine &&
+      change.range.endLineNumber <= endLine
+
+    const isDeletion = change.text == ''
+    let deletionFromBlock = false
```

Comment:
Just a suggestion as in the other PR about boolean names (unless you decided you're not into that, in which case, ignore this)

`hasChangeInBlock` 
`hasDeletionFromBlock`

Reply:
done